src/components/TextInput.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Field } from 'redux-form';
+import TextField from 'material-ui/TextField';
+import PropTypes from 'prop-types';
+
+function RenderTextField({ ...props }) {
+  const textProp = Object.assign({}, props);
+  delete textProp.meta;
+  delete textProp.input;
+
+  return (
+    <TextField
+      {...textProp}
+    />
+  );
+}
+const TextInput = ({ name, ...props }) => (
+  <Field
+    {...props}
+    name={name}
+    id={name}
```

Comment:
You are right after using Maperror, the warning doesn't come i was able reproduce it by passing the props again and deleting meta , input. I think somehow Maperror handels this in all cases this is the warning.
browser.js:49 Warning: Material-UI: We don't have enough information
 to build a robust unique id for the TextField component. Please provide an id or a name.